src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilderInstantiation.cs

```diff
@@ -1,22 +1,38 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Collections;
-using System.Globalization;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.InteropServices;
 
 namespace System.Reflection.Emit
 {
-    internal sealed class TypeBuilderInstantiation : TypeInfo
+    /*
+     * TypeBuilderInstantiation represents an instantiation of a generic TypeBuilder.
+     */
+#if MONO_FEATURE_SRE
+    [StructLayout(LayoutKind.Sequential)]
+#endif
+    internal sealed partial class TypeBuilderInstantiation : TypeInfo
     {
+#region Keep in sync with object-internals.h MonoReflectionGenericClass
+        private Type generic_type;
+        private Type[] type_arguments;
```

Comment:
```suggestion
        private Type _genericType;
        private Type[] _typeArguments;
```

Reply:
I kept them because they need to be in sync with mono native type, though I guess I could rename them in mono type too.